Bottom: 30f8ff8d229ecf7724823c8878a9023ef2771e47
Top:    26f7e817f2f2d6a10346230ac6323ed065b496d1
Author: Ian Kent <ikent@redhat.com>
Date:   2013-10-02 08:25:14 +0800

Refresh of autofs-5.0.7-add-std-vars-to-program-map-invokation.patch

---

diff --git a/modules/lookup_program.c b/modules/lookup_program.c
index 7e22b38..3e1e0af 100644
--- a/modules/lookup_program.c
+++ b/modules/lookup_program.c
@@ -272,6 +272,8 @@ int lookup_mount(struct autofs_point *ap, const char *name, int name_len, void *
 		if (ctxt->mapfmt && strcmp(ctxt->mapfmt, "MAPFMT_DEFAULT")) {
 			struct parse_context *pctxt = (struct parse_context *) ctxt->parse->context;
 			macro_setenv(pctxt->subst);
+			/* Add standard environment as seen by sun map parser */
+			ctxt->subst = addstdenv(ctxt->subst);
 		}
 		execl(ctxt->mapname, ctxt->mapname, name, NULL);
 		_exit(255);	/* execl() failed */
